### PR TITLE
bug: consider remaining bsp capacity for volunteering & confirming files

### PR DESCRIPTION
**Pull request by snowmead**

### CODE REVIEW

File: pallets/file-system/src/utils.rs [hunk 1]
```diff
@@ -428,6 +428,15 @@ where
...
         );
 
+        let available_capacity =
+            <T::Providers as ReadStorageProvidersInterface>::available_capacity(&bsp_id);
+
+        // Check if the BSP has enough capacity to store the file.
+        ensure!(
+            available_capacity > storage_request_metadata.size,
+            Error::<T>::InsufficientCapacity
+        );
```
Comment (TDemeco) [inline pallets/file-system/src/utils.rs:438]:
```suggestion
        // Check if the BSP has enough capacity to store the file.
        ensure!(
            available_capacity >= storage_request_metadata.size,
            Error::<T>::InsufficientCapacity
        );
```

File: pallets/file-system/src/utils.rs [hunk 2]
```diff
@@ -558,6 +567,15 @@ where
...
             );
 
+            let available_capacity =
+                <T::Providers as ReadStorageProvidersInterface>::available_capacity(&bsp_id);
+
+            // Check if the BSP has enough capacity to store the file.
+            ensure!(
+                available_capacity > storage_request_metadata.size,
+                Error::<T>::InsufficientCapacity
+            );
```
Comment (TDemeco) [inline pallets/file-system/src/utils.rs:577]:
```suggestion
            // Check if the BSP has enough capacity to store the file.
            ensure!(
                available_capacity >= storage_request_metadata.size,
                Error::<T>::InsufficientCapacity
            );
```

**Review by TDemeco — APPROVED** [review]
Looks good 👍🏻 minor changes

File: pallets/file-system/src/utils.rs (hunk 1 above)
Comment (snowmead) [reply pallets/file-system/src/utils.rs:438]: Yes, caught this.

**Pull request closed without merging**